### PR TITLE
Do not fail if user already member

### LINKED ISSUE #2772: User can not start exercise if participation status is repo_copied
### Describe the bug
Users participation had status repo created without build plan but was unable to start the exercise.

#### To Reproduce

1. Unknown

#### Expected behavior

There is code to handle the case that the user is already member of the repo but the behaviour of gitlab must have changed over time/versions.

```
artemis                    | 2021-01-28 17:52:30.668 ERROR 1 --- [io-8080-exec-30] o.z.problem.spring.common.AdviceTraits   : Internal Server Error
artemis                    | 
artemis                    | java.lang.NullPointerException: Cannot invoke "java.util.Map.containsKey(Object)" because the return value of "org.gitlab4j.api.GitLabApiException.getValidationErrors()" is null
artemis                    |    at de.tum.in.www1.artemis.service.connectors.gitlab.GitLabService.addMemberToRepository(GitLabService.java:129)
artemis                    |    at de.tum.in.www1.artemis.service.connectors.gitlab.GitLabService.configureRepository(GitLabService.java:113)
artemis                    |    at de.tum.in.www1.artemis.service.ParticipationService.configureRepository(ParticipationService.java:546)
artemis                    |    at de.tum.in.www1.artemis.service.ParticipationService.startProgrammingExercise(ParticipationService.java:255)
artemis                    |    at de.tum.in.www1.artemis.service.ParticipationService.startExercise(ParticipationService.java:219)
artemis                    |    at de.tum.in.www1.artemis.service.ParticipationService$$FastClassBySpringCGLIB$$f26b3f60.invoke(<generated>)
```

### Environment

gitlab

the code (GitLabService.java:129) expects a ValidationError in the Exeption when user already exist but Gitlab instead returns 409

### PR DESCRIPTION
**Pull request by 1-alex98**
Suggestion on how to fix / what to change concerning #2772. Tested locally.
Fixes #2772

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java [hunk 1]
```diff
@@ -101,17 +101,24 @@ public void addMemberToRepository(VcsRepositoryUrl repositoryUrl, User user) {
...
 
         try {
+            if (userAlreadyMember(repositoryId, userId)) return;
             gitlab.getProjectApi().addMember(repositoryId, userId, DEVELOPER);
+        } catch (GitLabApiException e) {
+            throw new GitLabException("Error while trying to add user to repository: " + user.getLogin() + " to repo " + repositoryUrl, e);
         }
-        catch (GitLabApiException e) {
-            if (e.getValidationErrors().containsKey("access_level")
-                    && e.getValidationErrors().get("access_level").stream().anyMatch(s -> s.contains("should be greater than or equal to"))) {
```
Comment (axel1200) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java:108]: check does not work is outdated

**Comment by axel1200** [comment]
This is supposed to be an idea on how to change current behavior

File: src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java (hunk 1 above)
Comment (axel1200) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java:108]: Turns out that is not true. But it only happens if you try to lower the access level.

Comment (axel1200) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java:108]: so this needs to stay as well

**Comment by ivanchimeno** [comment]
Pr #2929 adds a check which should prevent an exception from being thrown if the user is already member. Should I close this pr?

**Pull request closed without merging**